src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -295,9 +307,19 @@ def step(
                 " `v_prediction`"
             )
 
-        # 4. Clip "predicted x_0"
+        # 4. Clip or threshold "predicted x_0"
         if self.config.clip_sample:
-            pred_original_sample = torch.clamp(pred_original_sample, -1, 1)
+            pred_original_sample = torch.clamp(pred_original_sample, -self.config.sample_max_value, self.config.sample_max_value)
+        
+        if self.config.thresholding:
+            # Dynamic thresholding in https://arxiv.org/abs/2205.11487
+            dynamic_max_val = pred_original_sample \
```

Comment:
```suggestion
            pred_original_sample = self.apply_threshold(pred_orginal_sample)
```

Can we move this into it's own function? 

Then we can use the `# Copy` mechanism in all schedulers to copy the logic, e.g. as done here: https://github.com/huggingface/diffusers/blob/1b0b54a4264453818d7a6474d336a19922e3490a/src/diffusers/schedulers/scheduling_ddim.py#L31

This way everything stays in sync **and** is very readable :-)

Reply:
Note that the function should be part of the Scheduler so that it has access to `self.dynamic_thresholding_ratio`